src/test/java/com/fakng/fakngagrgtr/parser/AbstractParserTest.java

```diff
@@ -3,11 +3,34 @@
 import com.fakng.fakngagrgtr.persistent.company.Company;
 import com.fakng.fakngagrgtr.persistent.location.Location;
 import com.fakng.fakngagrgtr.persistent.vacancy.Vacancy;
+import org.apache.commons.io.IOUtils;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 public abstract class AbstractParserTest {
 
+    protected String getTestJson(String fileName) throws IOException {
+        return IOUtils.toString(Objects.requireNonNull(this.getClass().getResourceAsStream(fileName)), StandardCharsets.UTF_8);
+    }
+
+    protected WebClient mockWebClient(String body) {
+        return WebClient.builder()
+                .exchangeFunction(clientRequest ->
```

Comment:
Can lambda be moved to a separate method? And then personally I don't understand anything and it's very scary :)

Reply:
I don't think it will be a good method that returns another function in this specific case. It'll be even harder to understand. This one is actually readable, I think